meson.build

```diff
@@ -156,13 +159,21 @@ mandir = join_paths(prefix, get_option('mandir'))
 localedir = join_paths(prefix, get_option('localedir'))
 
 gio = dependency('gio-2.0', version : '>= 2.45.8')
-giounix = dependency('gio-unix-2.0', version : '>= 2.45.8')
+giounix = dependency('gio-unix-2.0', version : '>= 2.45.8', required: false)
+if giounix.found()
+  conf.set('HAVE_GIO_UNIX', '1')
+endif
 if gio.version().version_compare ('>= 2.55.0')
   conf.set('HAVE_GIO_2_55_0', '1')
 endif
 if build_standalone
 gmodule = dependency('gmodule-2.0')
-gudev = dependency('gudev-1.0', version : '>= 232')
+if get_option('gudev')
+  gudev = dependency('gudev-1.0', version : '>= 232')
+  conf.set('HAVE_GUDEV', '1')
+else
+  gudev = dependency('', required : false)
```

Comment:
Instead can you just += gudev in meson.build in subdirectory perhaps based on get_option? Empty thing like this seems wrong.

Reply:
I've seen other projects do it. Using a `null` dep means you don't have to have lots of temporary lists, and can just unconditionally include gudev in all cases. I tried without, and the meson.build files got a bit weird.

Reply:
OK.